activerecord/CHANGELOG.md

```diff
@@ -33,6 +33,13 @@
 
     *Ryuta Kamizono*
 
+
+*   Fix Collection cache key with limit and custom select (PG:AmbigousColumn: Error)
```

Comment:
This not only happen on postgres but also on mysql2, so preferable more generic changelog entry, and please add new changelog entry to top of the file.

```
% ARCONN=mysql2 be ruby -w -Itest test/cases/collection_cache_key_test.rb -n test_cache_key_for_relation_with_custom_select_and_limit
Using mysql2
Run options: -n test_cache_key_for_relation_with_custom_select_and_limit --seed 6336
# Running:
E

Error:ActiveRecord::CollectionCacheKeyTest#test_cache_key_for_relation_with_custom_select_and_limit:
ActiveRecord::StatementInvalid: Mysql2::Error: Duplicate column name 'updated_at': SELECT COUNT(*) AS `size`, MAX(subquery_for_cache_key.updated_at) AS timestamp FROM (SELECT developers.*, `developers`.`updated_at` FROM `developers` WHERE `developers`.`salary` = ? ORDER BY `devel
opers`.`updated_at` DESC LIMIT ?) subquery_for_cache_key
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb:116:in `prepare
'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb:116:in `block i
n exec_stmt_and_free'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:601:in `block (2 levels)
 in log'
    /Users/kamipo/.rbenv/versions/2.5.1/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:600:in `block in log'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb:591:in `log'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb:112:in `exec_st
mt_and_free'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb:39:in `exec_query'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:455:in `select'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:62:in `select_all'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb:101:in `select_all'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb:12:in `select_all'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:69:in `select_one'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/collection_cache_key.rb:35:in `collection_cache_key'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/relation.rb:297:in `cache_key'
    test/cases/collection_cache_key_test.rb:50:in `block in <class:CollectionCacheKeyTest>'


rails test test/cases/collection_cache_key_test.rb:45
```